### PR TITLE
Duplo 14623 Node pool Data Source

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Added datasource for node pool
## Summary of changes

This PR does the following:

- Added datasource `duplocloud_gcp_node_pools`
- Added datasource `duplocloud_gcp_node_pool`

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-14623 Terraform: GCP Node Pools](https://app.clickup.com/t/8655600/DUPLO-14623)

File: duplocloud/data_source_duplo_gcp_node_pools.go [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+
+func dataGcpK8NodePoolFunctionSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"tenant_id": {
+			Description:  "The GUID of the tenant that the node pool will be created in.",
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.IsUUID,
+		},
+		"name": {
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_node_pools.go:44]: Add `fullname` attribute to the schema and set proper value.

**Review by tahir-duplo — COMMENTED** [review]
Generate documentation using make doc.

File: duplocloud/data_source_duplo_gcp_node_pools.go (hunk 1 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_node_pools.go:44]: added

File: duplocloud/data_source_duplo_gcp_node_pools.go [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+			Description: "The short name of the node pool.",
+			Type:        schema.TypeString,
+			Computed:    true,
+			Optional:    true,
+		},
+		"fullname": {
+			Description: "The full name of the node pool.",
+			Type:        schema.TypeString,
+			Computed:    true,
+			Optional:    true,
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_node_pools.go:54]: Remove optional from `fullname`

File: duplocloud/data_source_duplo_gcp_sql.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	"terraform-provider-duplocloud/duplosdk"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceGCPCloudSQL() *schema.Resource {
+	return &schema.Resource{
+		Description: "`duplocloud_gcp_sql_cloud` retrieves a gcp sql instance in Duplo.",
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_sql.go:16]: Change `duplocloud_gcp_sql_cloud` to `duplocloud_gcp_sql_database_instance`

File: duplocloud/data_source_duplo_gcp_sqls.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	"terraform-provider-duplocloud/duplosdk"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceGCPCloudSQLs() *schema.Resource {
+	return &schema.Resource{
+		Description: "`duplocloud_gcp_sql_cloud` retrieves a gcp sql instance in Duplo.",
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_sqls.go:15]: Change `duplocloud_gcp_sql_cloud` to `duplocloud_gcp_sql_database_instances`

File: duplocloud/data_source_duplo_gcp_sqls.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+		Schema: map[string]*schema.Schema{
+			"tenant_id": {
+				Description:  "The GUID of the tenant that the sql database will be created in.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+			"sqls": {
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_sqls.go:27]: Rename field `sqls` to `databases`

File: duplocloud/data_source_duplo_gcp_node_pool.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	"log"
+	"terraform-provider-duplocloud/duplosdk"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceGCPNodePool() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceGCPNodePoolRead,
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_node_pool.go:15]: Add `Description:`

File: duplocloud/data_source_duplo_gcp_node_pools.go [hunk 7]
```diff
@@ -0,0 +1,420 @@
...
+	"terraform-provider-duplocloud/duplosdk"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceGCPNodePools() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceGCPNodePoolList,
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_node_pools.go:16]: Add Description:

File: duplocloud/data_source_duplo_gcp_node_pools.go (hunk 2 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_node_pools.go]: done

File: duplocloud/data_source_duplo_gcp_sql.go (hunk 3 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_sql.go]: done

File: duplocloud/data_source_duplo_gcp_sqls.go (hunk 4 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_sqls.go]: done

File: duplocloud/data_source_duplo_gcp_sqls.go (hunk 5 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_sqls.go]: done

File: duplocloud/data_source_duplo_gcp_node_pool.go (hunk 6 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_node_pool.go:17]: done

File: duplocloud/data_source_duplo_gcp_node_pools.go (hunk 7 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_node_pools.go:18]: done

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**